db-tests/src/test/java/gov/nasa/jpl/aerie/database/ExternalEventTests.java

```diff
@@ -0,0 +1,1922 @@
+       *    valid source.
+       *
+       *    A:  +++++++
+       *    B:     ++++++++
+       *        AAABBBBBBBB
+       */
+      @Test
+      void testBackwardOverlap() throws SQLException {
+        try(final var statement = connection.createStatement()) {
+
+          // create our sources and their per-window events
+          ExternalSource a = new ExternalSource(
+              "A",
+              st,
+              dg,
+              "2024-01-01T00:00:00Z",
+              "2024-01-01T00:00:00Z",
+              "2024-01-01T01:00:00Z",
+              ca,
+              mt
+          );
+          ExternalEvent aE = new ExternalEvent(
+              a.key() + "_event",
+              st,
+              a.key(),
+              dg,
+              "2024-01-01T00:00:00Z",
+              duration,
+              mt
+          ); // have to manually pick this
+          ExternalSource b = new ExternalSource(
+              "B",
+              st,
+              dg,
+              "2024-01-02T00:00:00Z",
+              "2024-01-01T00:30:00Z",
+              "2024-01-01T01:00:00Z",
+              ca,
+              mt
+          );
+          ExternalEvent bE = new ExternalEvent(b.key() + "_event", st, b.key(), dg, b.start_time(), duration, mt);
+
+          // verify the ranges are as expected
+          assertDoesNotThrow(() -> {
+            // insert generic external event type, source type, and derivation group
+            insertStandardTypes();
+
+            // insert sources
+            insertExternalSource(a);
+            insertExternalSource(b);
+
+            // insert events
+            insertExternalEvent(aE);
+            insertExternalEvent(bE);
+
+            var res = statement.executeQuery(
+                // language=sql
+                """
+                SELECT * FROM merlin.derived_events ORDER BY source_key
+                """
+            );
+
+            // verify the range for A is shorter than what is specified in the definition of "a" - it should end sooner.
+            final String[] expectedResults = {
+                "{[\"2024-01-01 00:00:00+00\",\"2024-01-01 00:30:00+00\")}",
+                "{[\"2024-01-01 00:30:00+00\",\"2024-01-01 01:00:00+00\")}"
+            };
+            compareLists(expectedResults, res, "source_range");
+          });
+        }
+      }
+
+      /*
+        A source in present before all sources, in a gap, should be ever present, even if chopped into several
+          subintervals:
+          A:      +++++++++++++++++++++
+          B:  ++++++
+          C:                +++++++
+              BBBBBBAAAAAAAACCCCCCCAAAA
+       */
+      /**
+       * This test is an overlapping case wherein there is a source preceding all sources in validity, that covers a
+       *    larger, more inclusive interval. This source should be ever present, even if chopped into several
```

Comment:
Please tweak wording, I can only understand what the case is because of the provided diagrams. (Thank you very much for providing these diagrams, by the way)

Reply:
Fixed.